src/main/java/bio/terra/cli/service/SamService.java

```diff
@@ -285,26 +223,20 @@ public static GroupPolicy fromSamPolicy(String samPolicyName) {
    * @param userEmail email of the user to add
    */
   public void addUserToGroup(String groupName, GroupPolicy policy, String userEmail) {
-    GroupApi groupApi = new GroupApi(apiClient);
-    try {
-      // - try to add the email to the group
-      // - if this fails with a Bad Request error, it means the email is not found
-      // - so try to invite the user first, then retry adding them to the group
-      callAndHandleOneTimeErrorWithRetries(
-          () -> {
-            groupApi.addEmailToGroup(groupName, policy.getSamPolicy(), userEmail);
-            return null;
-          },
-          SamService::isBadRequest,
-          () -> {
-            inviteUserNoRetries(userEmail);
-            return null;
-          });
-    } catch (ApiException | InterruptedException ex) {
-      throw new SystemException("Error adding user to SAM group.", ex);
-    } finally {
-      closeConnectionPool();
-    }
+    // - try to add the email to the group
+    // - if this fails with a Bad Request error, it means the email is not found
+    // - so try to invite the user first, then retry adding them to the group
+    handleClientExceptions(
```

Comment:
Yeah I didn't do this before because this pattern was only used once or twice, so it didn't save much repetition. But from a code readability standpoint, I see your point. Added helper methods that handle the one-time error and log the client exceptions.